samples/detect.js

```diff
@@ -52,43 +68,43 @@ function detectTextIntent(projectId, sessionId, queries, languageCode) {
       },
     };
 
-    if (!promise) {
-      // First query.
-      console.log(`Sending query "${query}"`);
-      promise = sessionClient.detectIntent(request);
-    } else {
-      promise = promise.then(responses => {
-        console.log('Detected intent');
-        const response = responses[0];
-        logQueryResult(sessionClient, response.queryResult);
-
-        // Use output contexts as input contexts for the next query.
-        response.queryResult.outputContexts.forEach(context => {
-          // There is a bug in gRPC that the returned google.protobuf.Struct
-          // value contains fields with value of null, which causes error
-          // when encoding it back. Converting to JSON and back to proto
-          // removes those values.
-          context.parameters = struct.encode(struct.decode(context.parameters));
-        });
-        request.queryParams = {
-          contexts: response.queryResult.outputContexts,
-        };
-
-        console.log(`Sending query "${query}"`);
-        return sessionClient.detectIntent(request);
-      });
+    // If we have contexts from previous interactions with the bot add them here
+    if (contexts && contexts.length > 0) {
+      request.queryParams = {
+        contexts: contexts,
+      };
     }
-  }
 
-  promise
-    .then(responses => {
-      console.log('Detected intent');
-      logQueryResult(sessionClient, responses[0].queryResult);
-    })
-    .catch(err => {
-      console.error('ERROR:', err);
-    });
+    const responses = await sessionClient.detectIntent(request);
+    return responses[0];
+  }
 
+  async function executeQueries(projectId, sessionId, queries, languageCode) {
```

Comment:
Is it possible to simplify this down into one async function. For quickstarts, we try to keep them as straightforward as possible.

Reply:
I guess we could, but...
From my point of view, it will be much easier for the user to reuse this example if you have a function that abstracts the integration with DialogFlow API (function `detectIntent`) and a separate function that handles the business logic (function `executeQueries` in this case).
One demonstrates how to integrate with the API and make a call to detect the intent of a text, where as the other function only demonstrates how to detect multiple intents (which, to be honest, it doesn't sound like a common use case for a chatbot...it might be a valid use case for batch processing maybe).
That way they can easily find the valuable part of the code without having to understand what we are trying to do with multiple queries in this case (it took me a few minutes to go through the old code to get the point and purpose of the example).
Thoughts?

Reply:
I sorta look at this quickstart from the python sample point of view. 

```
loop through queries
   call detect Intent
```

We don't expect our quickstarts to be used for actual production use, but as the sorta bare minimum / simplest sample to use with the API (Removing the loop is probably even better for that). But we still try to do what is best for each language. But I'll defer to your judgement as to the best approach for Node.js as I'm not a language expert.

Reply:
I agree that removing the loop will make the example a lot more simpler and easier to understand. I'm not even sure why we are looping through messages but I saw we are doing the same on all the code examples so I wanted to mimc the behaviour in the best way for NodeJs.
If you look at the examples for the other languages, for instance c#, they don't leverage the context to simulate a session or a stateful conversation. When I was refactoring the code it did take me a while to figure out what the ultimate intent of the code was (pass the context form intent 1 to the 2nd request).

Anyway. I believe that the new NodeJs example has everything a user will need to understand how to interact with Dialogflow using the NodeJs SDK.